nasl/nasl_cert.c

```diff
@@ -930,8 +936,36 @@ nasl_cert_query (lex_ctxt *lexic)
           retc->size = strlen (name);
         }
     }
+  else if (!strcmp (command, "public-key-algorithm-name"))
+    {
+      gnutls_datum_t datum;
+      gnutls_x509_crt_t cert = NULL;
+      int algo;
+      const char *algo_name;
+
+      datum.data =
+        (void *) ksba_cert_get_image (obj->cert, (size_t *) &datum.size);
+      if (!datum.data)
+        return NULL;
+      if (gnutls_x509_crt_init (&cert) != GNUTLS_E_SUCCESS)
+        return NULL;
+      if (gnutls_x509_crt_import (cert, &datum, GNUTLS_X509_FMT_DER)
+          != GNUTLS_E_SUCCESS)
+        return NULL;
+      if ((algo = gnutls_x509_crt_get_pk_algorithm (cert, NULL)) < 0)
+        {
+          g_message ("%s: Error getting the public key algorithm name.",
+                     __func__);
+          return NULL;
+        }
+      algo_name = gnutls_pk_algorithm_get_name (algo) ? g_strdup (gnutls_pk_algorithm_get_name (algo)) : g_strdup ("unknown");
+      retc = alloc_typed_cell (CONST_DATA);
+      retc->size = strlen (algo_name);
+      retc->x.str_val = algo_name;
+    }
   else if (!strcmp (command, "modulus"))
     {
+      
```

Comment:
This also looks superfluous and could be dropped as none of the other `else if` parts are using a space after the `{`.